sharding/indexHashedNodesCoordinatorWithRater.go

```diff
@@ -0,0 +1,142 @@
+package sharding
+
+import (
+	"fmt"
+	"github.com/ElrondNetwork/elrond-go/core"
+)
+
+type indexHashedNodesCoordinatorWithRater struct {
+	indexHashedNodesCoordinator
+	RatingReader
+}
+
+// NewIndexHashedNodesCoordinator creates a new index hashed group selector
+func NewIndexHashedNodesCoordinatorWithRater(arguments ArgNodesCoordinator, rater RatingReader) (*indexHashedNodesCoordinatorWithRater,
+	error) {
+	indexNodesCoordinator, err := NewIndexHashedNodesCoordinator(arguments)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if rater == nil {
+		return nil, ErrNilRater
+	}
+
+	return &indexHashedNodesCoordinatorWithRater{
+		indexHashedNodesCoordinator: *indexNodesCoordinator,
+		RatingReader:                rater,
+	}, nil
+}
+
+func (ihgs *indexHashedNodesCoordinatorWithRater) ComputeValidatorsGroup(
```

Comment:
lot of duplicated code with nodehashedcoordinator.

Reply:
yes... because there are no virtual methods, the call to ComputeValidatorsGroup has to be made every time from a method inside the indexHashedNodesCoordinatorWithRater... I'm still thinking about a solution.

Reply:
refactored, and set a single method in the nodesCoordinator to be overwritten... not the best solution though